Dapper/SqlMapper.cs

```diff
@@ -3379,7 +3383,7 @@ private static void GenerateDeserializerFromMap(Type type, IDataReader reader, i
             }
             il.MarkLabel(allDone);
             il.BeginCatchBlock(typeof(Exception)); // stack is Exception
-            il.Emit(OpCodes.Ldloc_0); // stack is Exception, index
+            il.Emit(OpCodes.Ldloc, currentIndexDiagnosticLocal); // stack is Exception, index

```

Comment:
I'm going to submit a PR after this one which requires a LocalBuilder instance to be passed to each stloc and ldloc to make it easier to follow what's happening and to add guardrails for future refactoring.
It would have been hard to ignore the mistake in my first PR if LocalBuilders were passed around instead of using magic indexes without passing them around.

Reply:
https://github.com/StackExchange/Dapper/pull/1282